### ISSUE TITLE
Add more language models

### ISSUE THREAD

**Issue by dzlab** [OpenMined/SyferText#4]
Currently only `en_core_web_lg` is availble to load and it has a size of 860MB which are downloaded the first time we do:
```python
nlp = syfertext.load('en_core_web_lg', owner = me)
```
Also, the models are hosted on github https://github.com/AlanAboudib/syfertext_en_core_web_lg/ may be this has to be hosted somewhere else

**Comment by AlanAboudib** [comment]
I raised the priority of this issue in the project todos. A lightweight model is important to have for CI.

**Comment by AlanAboudib** [comment]
We should add 'en_core_web_sm' https://spacy.io/models/en

This should be the model to use for tests in CI

**Comment by dzlab** [comment]
the sm model is 11Mb which is still big for a CI, for now I think it is OK to use but later maybe it will be worth training a super small model https://spacy.io/usage/training

**Comment by AlanAboudib** [comment]
@dzlab 11Mb is actually ok for CI given that it is downloaded only once and reused. However, it all depends on what we are going to test. It would also be a good idea to create a dummy language model that contains no real vocabulary.

**Comment by shashank-m** [comment]
Hey can I start working on this issue?

**Comment by AlanAboudib** [comment]
@shashank-m sure you can

**Comment by sachin-101** [comment]
@AlanAboudib I would like to work on this issue. I will go ahead with adding the functionality to load the above-mentioned model of size 11Mb.

**Comment by AlanAboudib** [comment]
Hi Guys, @shashank-m @sachin-101  I prefer you work on adding two models:

en_core_sci_md
en_core_sci_lg

From here

https://allenai.github.io/scispacy/

Please contact me on slack to discuss this further

**Comment by sachin-101** [comment]
Ok. On it.

**Comment by dzlab** [comment]
@AlanAboudib it is a bad design to download models from github like this: https://github.com/AlanAboudib/syfertext_en_core_web_lg/blob/master/key2row?raw=true
We should build a functionality to download directely from spacy, e.g.  [with pip](https://spacy.io/usage/models#download-pip) or [via releases](https://spacy.io/usage/models#download-manual). We can add this step to the `setup.py` but may be. this should go into a different PR?

**Comment by AlanAboudib** [comment]
@dzlab That is true. This is a temporary solution. For the moment SyferText language models are 'peeled down' from spaCy's language models. They are not directly compatible. The files we use are not exactly the same as the ones stored in spaCy's language model. For instance, for the 'en_core_web_sm' the vectors are not explicitly provided.

spaCy language models are python modules that are packaged. They can be loaded as Language objects directly. This mechanism is not yet available in SyferText but it is part of the roadmap I announced earlier:

https://github.com/OpenMined/Roadmap/tree/master/nlp_team

in short-term goal, 4).

A prerequisite to being able to do that is to figure out the pipelining/subpipelining mechanism that is typically stored as part of the language model. Until then, we are going to use those files in direct download. I am almost done of implementing subpipelines. We will discuss it soon.

I asked @sachin-101 to perform the tests after loading the files to his repo. Once the PR is reviewed, we might consider storing the files as a submodule of SyferText

**Comment by ShivamSRS** [comment]
i am confused by this issue. Do you want to add new language models like bert, gpt or xlnet? if so i am interested in implementing them

**Comment by AlanAboudib** [comment]
Hi @ShivamSRS thank you for your interest. Adding language model here is in the sense of adding other models than `en_core_web_lg` that we are currently using. This issue was addressed by @sachin-101 in the PR #35 but now it is put on hold waiting for the merge of the PyGrid integration Epic I am working on here #93 

I am working on the merge now. After I do, MUCH work will be needed to add such language models. So any contribution with be welcome :)

**Comment by AlanAboudib** [comment]
I will close this issue. We will deal with adding language models when syfertext 0.1.0 is released
